elpy-shell.el

```diff
@@ -146,6 +146,25 @@ the code cell beginnings defined here."
 (defvar elpy--shell-last-py-buffer nil
   "Help keep track of python buffer when changing to pyshell.")
 
+(defvar elpy-shell--python-version-cache nil)
+(defvar elpy-shell--python-interpreter-cache nil)
```

Comment:
Maybe this is poorly named. The var just keeps track of the value of `python-shell-interpreter` from the last time `elpy-shell-get-python-version` was run so that if the value of `python-shell-interpreter` has changed then we can make a call to get the new python version. So it's not really a "cache" in the sense of a performance improvement by storing some value we don't expect to change often.

Reply:
I am personally okay with it, as long as the docstring of the variable explains what it is properly.

Or maybe `elpy-shell--python-version-cached-interpreter' ?